wordpress-mobile/WordPress-Android
# Drop jetifier

Merge Instructions:
1. ~Review and approve (don't merge) - https://github.com/wordpress-mobile/gutenberg-mobile/pull/4361~
2. ~Review this PR~
3. ~Merge Aztec https://github.com/wordpress-mobile/AztecEditor-Android/pull/946/files~
4. ~Merge Stories https://github.com/Automattic/stories-android/pull/717~
5. ~Merge GB PR - https://github.com/wordpress-mobile/gutenberg-mobile/pull/4361~
6. ~Merge GB PR which updates Aztec Version - it doesn't exist yet~
7. ~Update GB hash/version in this repo~
8. ~Review and Merge upgrade of test dependencies - https://github.com/wordpress-mobile/WordPress-Android/pull/16170 (technically not a blocker)~
9. ~Merge the branch in .mobile-secrets into trunk~
10. ~Update hash and branch in .configure~ (9f811ea608958fb7acbe781672b4c4c91318c5d4)
11. ~Remove "Not ready for merge" label~
12. ~Merge this PR~
13. Optional: Ping the team that they can change `android.enableJetifier=true` to `android.enableJetifier=false` in their gradle.properties or run `./gradlew configureApply`.

This PR migrates all libraries to versions which are migrated to AndroidX so we can disable jetifier on our builds. Jetifier was always intended to be used only temporarily and is planned to be removed from future versions of AGP.


List of changes
- Bumps Stetho version to 1.6.0 - the only change is that the version is migrated to AndroidX - https://github.com/facebook/stetho/releases
- Bumps ExoPlayer from 2.9.3 to 2.12.3 -> this was the most challenging task as ExoPlayer is used in both MediaPreivewFragment and Gutenberg-Mobile => we had to bump the version in both Gutenberg and MediaPreviewFragment. (This work isn't done yet, the GB part is still WIP)
- Updates WordPress utils version to a version which is migrated to AndroidX -> I bumped the version in all the modules to make the version consistent
- Adds our S3 repo to Analytics and Networking modules so the new version of WordPress utils is reachable
- Disables jetifier - updated gradle.properties and .configure
- "Manually" (using cmd tool) Jetified the Tenor library since it hasn't been updated since 2018 and I don't think it'll get updated any time soon
- Updated Aztec version with version that uses the updated version of WordPress utils

To test:
Make sure to disable Jetifier in your local gradle.properties  => Smoke test the app - anything you can think of.


1. Open Media Library
2. Open a detail of a video
3. Tap on the preview
4. Make sure the preview is displayed correctly and you can play the video
-------------------
1. Create a new post
2. Add image block
3. Select "Choose from Tenor"
4. Type "Cat"
5. Add a gif
---------------
1. Create a new post 
2. Add Cover block
3. Add Video
4. Choose a video file
5. Make sure the video preview works


## Regression Notes
1. Potential unintended areas of impact
- Video previews (media library)
- Video blocks in Gutenberg 
- Adding gifs from Tenor

2. What I did to test those areas of impact (or what existing automated tests I relied on)
The above "to test" section

3. What automated tests I added (or what prevented me from doing so)

None - this is a technical improvement which doesn't change the UX

PR submission checklist:
